tests/unit/postTestBuildComment.js

```diff
@@ -0,0 +1,61 @@
+const {when} = require('jest-when');
+
+const core = require('@actions/core');
+const GithubUtils = require('../../.github/libs/GithubUtils');
+
+const mockGetInput = jest.fn();
+const mockCreateComment = jest.fn();
+
+jest.mock('@actions/github', () => ({
+    context: {
+        repo: {
+            repo: 'repo',
+        },
+    },
+}));
+
+const ghAction = require('../../.github/actions/javascript/postTestBuildComment/postTestBuildComment');
+
+const androidLink = 'https://expensify.app/ANDROID_LINK';
+const iOSLink = 'https://expensify.app/IOS_LINK';
+const webLink = 'https://expensify.app/WEB_LINK';
+const desktopLink = 'https://expensify.app/DESKTOP_LINK';
+
+const androidQRCode = `![Android](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=${androidLink})`;
+const desktopQRCode = `![iOS](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=${desktopLink})`;
+const iOSQRCode = `![iOS](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=${iOSLink})`;
+const webQRCode = `![iOS](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=${webLink})`;
+
+let message = ':test_tube::test_tube: Use the links below to test this build in android and iOS. Happy testing! :test_tube::test_tube:';
+message += '\n| android :robot:  | iOS :apple: |';
+message += '\n| ------------- | ------------- |';
+message += `\n| ${androidLink}  | ${iOSLink}  |`;
+message += `\n| ${androidQRCode}  | ${iOSQRCode}  |`;
+message += '\n| desktop :computer: | web :spider_web: |';
+message += `\n| ${desktopLink}  | ${webLink}  |`;
+message += `\n| ${desktopQRCode}  | ${webQRCode}  |`;
+
+describe('Post test build comments action tests', () => {
+    beforeAll(() => {
+        // Mock core module
+        core.getInput = mockGetInput;
+        GithubUtils.createComment = mockCreateComment;
+    });
+
+    test('Test GH action', () => {
+        when(core.getInput).calledWith('PR_NUMBER', {required: true}).mockReturnValue(12);
```

Comment:
Interesting pattern I haven't seen before. I'm not strongly opposed because this looks like a pretty clean approach. It would be nice to have some guidance for when we should use these "partial mocks". Also trying to understand what the alternative would look like. Would it be something like this?

```js
mockGetInput.mockImplementation((input) => {
    switch(input) {
        case 'PR_NUMBER':
            return 'success';
        case 'IOS':
            return 'success';
        ...
        ...
        default:
            throw new Error(`No value for ${input}`);
    }
})
```

Reply:
AFAIK that is the alternative.

Partial mock is useful when we expect method to be invoked several times in one test with different parameters and do not want to rely of order of those executions.

Reply:
I think that this is straightforward enough to read that I won't push back on using `when` here